### ISSUE TITLE
Starting two instances with auto port scan fails

### ISSUE THREAD

**Issue by ndac-todoroki** [thecoshman/http#88]
### What occurs
```
$ httplz &
Hosting "." on port 8000 without TLS and no authentication...
Ctrl-C to stop.

$ httplz
Starting server failed.
```

Doing `http(lz) -p 8001` works, but I need to know all the ports I am using.

### What I expected 
The second command should start the server at port 8001.

EDIT: or at least more friendly errors telling me the reason of the failure; It is different from the error when doing `http -p 8000`

### Environments
macOS 10.14.6
http 1.5.1, build with `cargo install`

**Comment by nabijaczleweli** [comment]
You're right, the error handling is Not Great. Can you try with the previous commit on the `develop` branch and post what the error is? I can't reproduce this

**Comment by nabijaczleweli** [comment]
hmm I bet it's the `if !error_s.contains("port")` check, this should be easy to fix if OSX's error reporting is sensible

**Comment by ndac-todoroki** [comment]
Here's the output:
```
httplz on  HEAD is 📦 v1.5.1 via 🦀 v1.37.0
❯ cargo build --release

❯ target/release/httplz &
Hosting "." on port 8000 without TLS and no authentication...
Ctrl-C to stop.

✦ ❯ target/release/httplz
Starting server failed: Address already in use (os error 48).
```

This means the port scan is not working well on OSX?

**Comment by nabijaczleweli** [comment]
Indeed, but if you would be so kind so as to try with the new commit on `develop`, this error message should be accepted as "port taken" instead of "fatal error"?

**Comment by ndac-todoroki** [comment]
For 682d1ac...
```
❯ target/release/http &
Hosting "." on port 8000 without TLS and no authentication...
Ctrl-C to stop.

httplz on  HEAD is 📦 v1.5.1 via 🦀 v1.37.0
✦ ❯ target/release/http
Hosting "." on port 8001 without TLS and no authentication...
Ctrl-C to stop.
```
Succeeded! 🥇:tada:

**Comment by nabijaczleweli** [comment]
Released in [`v1.5.2`](https://github.com/thecoshman/http/releases/tag/v1.5.2)
